test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -68,13 +68,23 @@ public ResultValidator expectEvents(Object... expectedEvents) {
         Iterator<EventMessage<?>> iterator = publishedEvents.iterator();
         for (Object expectedEvent : expectedEvents) {
             EventMessage actualEvent = iterator.next();
-            if (EventMessage.class.isInstance(expectedEvent)) {
-                EventMessage<?> expectedEventMessage = (EventMessage<?>) expectedEvent;
-                if (!verifyEventEquality(expectedEventMessage.getPayload(), actualEvent.getPayload())
-                        || !verifyMetaDataEquality(expectedEventMessage.getPayloadType(), expectedEventMessage.getMetaData(), actualEvent.getMetaData())) {
-                    reporter.reportWrongEvent(publishedEvents, Arrays.asList(expectedEvents), actualException);
-                }
-            } else if (!verifyEventEquality(expectedEvent, actualEvent.getPayload())) {
+            if (!verifyEventEquality(expectedEvent, actualEvent.getPayload())) {
+                reporter.reportWrongEvent(publishedEvents, Arrays.asList(expectedEvents), actualException);
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public ResultValidator expectEvents(EventMessage... expectedEvents) {
+        this.expectEvents((Object[]) expectedEvents);
+
+        Iterator<EventMessage<?>> iterator = publishedEvents.iterator();
+        for (EventMessage expectedEvent : expectedEvents) {
+            EventMessage actualEvent = iterator.next();
+            if (!verifyMetaDataEquality(expectedEvent.getPayloadType(),
```

Comment:
Don't you intent to verify both the event equality (by using `verifyEventEquality()`) and the meta data equality (by leveraging `verifyMetaDataEquality()`) in the `ResultValidator expectEvents(EventMessage...)` function?

Reply:
@smcvb see line 80. There is a call an explicit call to `expectEvents(Object... expectedEvents)`.

I think this clearly conveys the overloaded method does whatever the original method does, plus `verifyMetaDataEquality()`.

Reply:
..fully missed that, guessing the `this` on that line put me off or something. 
Thanks for pointing that out, I agree with your second statement there.